nexus/reconfigurator/execution/src/datasets.rs

```diff
@@ -282,4 +423,250 @@ mod tests {
             nzones_with_durable_datasets + 2,
         );
     }
+
+    #[nexus_test]
+    async fn test_dataset_records_update(cptestctx: &ControlPlaneTestContext) {
+        const TEST_NAME: &str = "test_dataset_records_update";
+
+        // Set up.
+        let nexus = &cptestctx.server.server_context().nexus;
+        let datastore = nexus.datastore();
+        let opctx = OpContext::for_tests(
+            cptestctx.logctx.log.clone(),
+            datastore.clone(),
+        );
+        let opctx = &opctx;
+
+        // Use the standard example system.
+        let (_, _, blueprint) = example(&opctx.log, TEST_NAME, 5);
+
+        // Record the sleds and zpools.
+        crate::tests::insert_sled_records(datastore, &blueprint).await;
+        crate::tests::create_disks_for_zones_using_datasets(
+            datastore, opctx, &blueprint,
+        )
+        .await;
+
+        let mut all_datasets = get_all_datasets_from_zones(&blueprint);
+        let EnsureDatasetsResult { inserted, updated, removed } =
+            ensure_dataset_records_exist(opctx, datastore, all_datasets.iter())
+                .await
+                .expect("failed to ensure datasets");
+        assert_eq!(inserted, all_datasets.len());
+        assert_eq!(updated, 0);
+        assert_eq!(removed, 0);
+
+        // These values don't *really* matter, we just want to make sure we can
+        // change them and see the update.
+        let first_dataset = &mut all_datasets[0];
+        assert_eq!(first_dataset.quota, None);
+        assert_eq!(first_dataset.reservation, None);
+        assert_eq!(first_dataset.compression, "");
+
+        first_dataset.quota = Some(ByteCount::from_kibibytes_u32(1));
+        first_dataset.reservation = Some(ByteCount::from_kibibytes_u32(2));
+        first_dataset.compression = String::from("lz4");
+        let _ = first_dataset;
```

Comment:
What is this line doing?

Reply:
I think this was previously helping me avoid holding a mutable borrow for too long, but it doesn't seem necessary anymore. Removed (and also for some other uses of `let _ = ...` below)